src/sagemaker_mxnet_container/distributed.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+from contextlib import contextmanager
+import logging
+import os
+import socket
+import subprocess
+
+from retrying import retry
+
+logger = logging.getLogger(__name__)
+
+
+class DefaultParameterServer():
+    ROLES = ['worker', 'scheduler', 'server']
+
+    def __init__(self, hosts, ps_port='8000', ps_verbose='0'):
+        self.scheduler = self.scheduler_host(hosts)
+        self.hosts = hosts
+        self.ps_port = ps_port
+        self.ps_verbose = ps_verbose
+
+    def scheduler_host(self, hosts):
+        return sorted(hosts)[0]
```

Comment:
is it necessary to sort this? can we just just the first one? Do all the hosts get this list in the same order?

Reply:
Also this function does not have unit tests. I think we should tests all public functions.

Reply:
it's probably not necessary, but the user could theoretically override the environment variable and give the list in a different order each time?

added a test. also made the function private because I remembered that I wanted it to be "memoized" (it's not proper memoization, I know, I know...) in `self.scheduler` anyway.

Reply:
i guess nobody is training with millions of hosts. The over head of sorting it is probably ok.